academic-requirements/src/components/FourYearPlanPage.tsx

```diff
@@ -8,57 +15,120 @@ const FourYearPlanPage = (props: {
+    //props.onClickCategory(category);
+    //setConcentrationOptions(concentrations); May be able to delete this line.
+  }
+
+  // RemoveDuplicates function.
+  function RemoveDuplicates(strings: string[]): string[] {
+    //Push all strings to a set(which disallows duplicates)
+    let set = new Set<string>();
+    strings.forEach((x) => {
+      set.add(x);
+    });
+    //Reassign all strings in the set to an array.
+    let arr = new Array<string>;
+    set.forEach((x) => {
+      arr.push(x);
+    });
+    //Return the array.
+    return arr;
+  }
+  //extractCategories function.
+  function extractCategories() {
+    //Initialize new array.
+    let i = new Array<string>();
+    //map is what loops over the list
+    //map calls arrow function, runs whats between curly braces.
+    //Push course categories from major and concentration course lists to array.
+    props.majorCourseList.map((course, index) => { i.push(course.category) })
+    props.concentrationCourseList.map((course, index) => { i.push(course.category) })
+    //Remove duplicate categories from the array.
+    setCategories(RemoveDuplicates(i))
+  }
+
   return (
     <div>
       {props.showing && (
         <div className="screen">
           <div className="four-year-plan" data-testid="scheduleContent">
             <h1>Academic Planner</h1>
+
+
           </div>
           <ErrorPopup
             onClose={popupCloseHandler}
             show={visibility}
             title="Error"
             error={error}
           />
-          <div className="grid-container">
-            <div className="semesters-container">
-              <div className="grid-item">Semester 1</div>
-              <div className="grid-item">Semester 2</div>
-              <div className="grid-item">Semester 3</div>
-              <div className="grid-item">Semester 4</div>
-              <div className="grid-item">Semester 5</div>
-              <div className="grid-item">Semester 6</div>
-              <div className="grid-item">Semester 7</div>
-              <div className="grid-item">Semester 8</div>
-            </div>
-            <div className="class-dropdown">
-              {props.majorCourseList.map((course, index) => {
-                return <div key={index}>{course.name}</div>;
-              })}
-              {props.concentrationCourseList.map((course, index) => {
-                return <div key={index}>{course.name}</div>;
-              })}
-            </div>
+
+          
+                {/* { <div className="courseDropdowns">
+
+                <SearchableDropdown
+                  options={categories}
+                  label="Category"
+                  onSelectOption={selectedCategory} //If option chosen, selected Category activated.
+                  showDropdown={true}
+                  thin={true}
+                />
+              </div> }   */}
+
+          <div className="page">
+           <Example
+            PassedCourseList = {props.majorCourseList}
```

Comment:
I could be wrong, but I believe this is why all the categories are not being displayed. Only the major-related courses are being passed into the DropTarget component (through Example component) for the category dropdown so only major related categories are being sent to the category dropdown.

Reply:
Yup, that's why. That one is on me, it should be a quick fix but we should have a discussion about whether there's a reason to pass the concentration, major, and gen ed courses as three different lists because it makes the code more complicated and off the top of my head I don't know why we would need them to be separate.